joomla/joomla-cms
# [BS2] Fix for XSS issue with data-target attributes in BS2 JS

### Summary of Changes

This patch by @SniperSister fixes an publicly known XSS Problem in the BS2 JS reported to the JSST by @C-Lodder. As this issue is publicly known and can also impact 3rd partys the JSST decided to patch it in the public tracker. This also allows an wider group of people to test this patch.

### Testing Instructions

Make sure the following bootstrap js components still work
* Alert
* Carousel
* Collapse
* Dropdown
* Modal
* Scrollspy
* Tab

Try using the data-target and href methods and also try to verify that the mentioned components still work in the Joomla backend.

### Expected result

BS JS Code still works

### Actual result

The current JS Code has an know and publicly documented XSS Problem

### Documentation Changes Required

none

cc @joomla/security @mbabker
